dspace-api/src/main/java/org/dspace/storage/bitstore/BitstreamStorageManager.java

```diff
@@ -561,150 +563,155 @@ public static void delete(Context context, int id) throws SQLException
+            if(duplicateBitRow == null)
+            {
+                success = file.delete();
 
-                    String message = ("Deleted bitstream " + bid + " (file "
+                String message = ("Deleted bitstream " + bid + " (file "
                                 + file.getAbsolutePath() + ") with result "
                                 + success);
-                    if (log.isDebugEnabled())
-                    {
-                        log.debug(message);
-                    }
-                    if (verbose)
-                    {
-                        System.out.println(message);
-                    }
-
-                    // if the file was deleted then
-                    // try deleting the parents
-                    // Otherwise the cleanup script is set to
-                    // leave the db records then the file
-                    // and directories have already been deleted
-                    // if this is turned off then it still looks like the
-                    // file exists
-                    if( success )
-                    {
-                        deleteParents(file);
-                    }
+                if (log.isDebugEnabled())
+                {
+                    log.debug(message);
+                }
+                if (verbose) {
+                    System.out.println(message);
                 }
 
-                // Make sure to commit our outstanding work every 100
-                // iterations. Otherwise you risk losing the entire transaction
-                // if we hit an exception, which isn't useful at all for large
-                // amounts of bitstreams.
-                commitCounter++;
-                if (commitCounter % 100 == 0)
-                {
-                	System.out.print("Committing changes to the database...");
-                    context.commit();
-                    System.out.println(" Done!");
+                // if the file was deleted then
+                // try deleting the parents
+                // Otherwise the cleanup script is set to
+                // leave the db records then the file
+                // and directories have already been deleted
+                // if this is turned off then it still looks like the
+                // file exists
+                if (success) {
+                    deleteParents(file);
                 }
             }
-
-            context.complete();
         }
         // Aborting will leave the DB objects around, even if the
-        // bitstreams are deleted. This is OK; deleting them next
-        // time around will be a no-op.
-        catch (SQLException sqle)
-        {
+        // bitstreams are deleted. This is OK; they will be deleted
+        // next time around
+        catch (SQLException sqle) {
+            System.err.println("Error: " + sqle.getMessage());
             if (verbose)
```

Comment:
While you are at it, why not remove the condition test here and always report the exception regardless of 'verbose'?  That would take care of DS-3074 as well.  It is strange that we don't report an event which aborts the command.